spec/haproxy/templates/haproxy_config/rate_limit_spec.rb

```diff
@@ -0,0 +1,100 @@
+          'window_size' => '10s',
+          'table_size' => '10m'
+        }
+      }
+    end
+
+    let(:temp_properties) do # temp_properties is required since we cannot deep_merge :properties in its own let/assignment block
+      default_properties.deep_merge(request_limit_base_properties)
+    end
+
+    let(:properties) { temp_properties }
+
+    it 'sets up stick-tables' do
+      expect(backend_req_rate).to include('stick-table type ip size 10m expire 10s store http_req_rate(10s)')
+    end
+
+    it 'tracks requests in stick tables' do
+      expect(frontend_http).to include('tcp-request content track-sc1 src table st_http_req_rate')
+      expect(frontend_https).to include('tcp-request content track-sc1 src table st_http_req_rate')
+    end
+
+    context 'when requests_rate_limit.block is also provided' do
+      let(:properties) do
+        temp_properties.deep_merge({ 'requests_rate_limit' => { 'block' => 'true' } })
+      end
+
+      it 'adds http-request deny condition to http-in and https-in frontends' do
+        expect(frontend_http).to include('http-request deny status 429 content-type "text/plain" string "429: Too Many Requests" if { sc_http_req_rate(1) gt 5 }')
+        expect(frontend_https).to include('http-request deny status 429 content-type "text/plain" string "429: Too Many Requests" if { sc_http_req_rate(1) gt 5 }')
+      end
+    end
+  end
+
+  context 'when ha_proxy.connections_rate_limit properties "connections", "window_size", "table_size" are provided' do
+    let(:backend_conn_rate) { haproxy_conf['backend st_tcp_conn_rate'] }
+
+    let(:connection_limit_base_properties) do
+      {
+        'connections_rate_limit' => {
+          'connections' => '5',
+          'window_size' => '10s',
+          'table_size' => '10m'
+        }
+      }
+    end
+
+    let(:temp_properties) do # temp_properties is required since we cannot deep_merge :properties in its own let/assignment block
+      default_properties.deep_merge(connection_limit_base_properties)
+    end
+
+    let(:properties) { temp_properties }
+
+    it 'sets up stick-tables' do
+      expect(backend_conn_rate).to include('stick-table type ip size 10m expire 10s store conn_rate(10s)')
+    end
+
+    it 'tracks connections in stick tables' do
+      expect(frontend_http).to include('tcp-request connection track-sc0 src table st_tcp_conn_rate')
+      expect(frontend_https).to include('tcp-request connection track-sc0 src table st_tcp_conn_rate')
+    end
+
+    context 'when connections_rate_limit.block is also provided' do
+      let(:properties) do
+        temp_properties.deep_merge({ 'connections_rate_limit' => { 'block' => 'true' } })
+      end
+
+      it 'adds http-request deny condition to http-in and https-in frontends' do
+        expect(frontend_http).to include('tcp-request connection reject if { sc_conn_rate(0) gt 5 }')
+        expect(frontend_https).to include('tcp-request connection reject if { sc_conn_rate(0) gt 5 }')
```

Comment:
Should this then also include the properties from 'tracks connections in stick tables'?

```suggestion
        expect(frontend_http).to include('tcp-request connection reject if { sc_conn_rate(0) gt 5 }')
        expect(frontend_https).to include('tcp-request connection reject if { sc_conn_rate(0) gt 5 }')
        
        expect(frontend_http).to include('tcp-request connection track-sc0 src table st_tcp_conn_rate')
        expect(frontend_https).to include('tcp-request connection track-sc0 src table st_tcp_conn_rate')
```

Reply:
Same point in lines 55-56